proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java

```diff
@@ -85,14 +85,22 @@ private static List<ReportPoint> fromOtlpRequest(
       OTLP_DATA_LOGGER.finest(() -> "Inbound OTLP Resource: " + resource);
       Pair<String, List<KeyValue>> sourceAndResourceAttrs =
           OtlpTraceUtils.sourceFromAttributes(resource.getAttributesList(), defaultSource);
-      String source = sourceAndResourceAttrs._1;
-      List<KeyValue> resourceAttributes =
-          includeResourceAttrsForMetrics ? sourceAndResourceAttrs._2 : Collections.EMPTY_LIST;
+      String source;
+      List<KeyValue> resourceAttributes;
 
       for (ScopeMetrics scopeMetrics : resourceMetrics.getScopeMetricsList()) {
         OTLP_DATA_LOGGER.finest(
             () -> "Inbound OTLP Instrumentation Scope: " + scopeMetrics.getScope());
         for (Metric otlpMetric : scopeMetrics.getMetricsList()) {
+
+          List<KeyValue> attrList = new ArrayList<KeyValue>(Collections.EMPTY_LIST);
+
+          updateAttrsListForOtelMetrics(resource, otlpMetric.getName(), attrList);
+
+          source = sourceAndResourceAttrs._1;
+          resourceAttributes =
+              includeResourceAttrsForMetrics ? sourceAndResourceAttrs._2 : attrList;
```

Comment:
Do you have to spend time populating attrList if includeResourceAttrsForMetrics is true?  Line 98 gets called to populate attrList either way. Consider rewriting updateAttrsListForOtelMetrics to return a List<KeyValue> Then call it directly here.

Reply:
make sense.

Reply:
Consider saying ```resourceAttributes = includeResourceAttrsForMetrics ? sourceAndREsourceAttrs._2 : updateAttrsListForOtelMetrics(resource, otlpMetric.getName())```  The advantage of doing it this way is that updateAttrsListForOtelMetrics only gets called when includeResourceAttrsForMetrics is false.